source/initial_temperature/continental_geotherm.cc

```diff
@@ -0,0 +1,257 @@
+
+    }
+
+
+
+    template <int dim>
+    void
+    ContinentalGeotherm<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Initial temperature model");
+      {
+        prm.enter_subsection("Continental geotherm");
+        {
+          prm.declare_entry ("Layer thicknesses", "30000.",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of thicknesses for the bottom of the lithospheric layers,"
+                             "for a total of N+1 values, where N is the number of compositional fields."
+                             "If only one value is given, then all use the same value.  Units: $kg / m^3$");
+          prm.declare_entry ("Surface temperature", "273.15",
+                             Patterns::Double (0),
+                             "The value of the surface temperature. Units: Kelvin.");
+          prm.declare_entry ("LAB isotherm temperature", "1673.15",
+                             Patterns::Double (0),
+                             "The value of the isothermal boundary temperature assumed at the LAB "
+                             "and up to the reference depth . Units: Kelvin.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    ContinentalGeotherm<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      unsigned int n_fields = 0;
+      prm.enter_subsection ("Compositional fields");
+      {
+        n_fields = prm.get_integer ("Number of fields");
+      }
+      prm.leave_subsection();
+
+
+      prm.enter_subsection ("Initial temperature model");
+      {
+        prm.enter_subsection("Continental geotherm");
+        {
+          LAB_isotherm = prm.get_double ("LAB isotherm temperature");
+          T0 = prm.get_double ("Surface temperature");
+          thicknesses = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Layer thicknesses"))),
+                                                                3,
+                                                                "Layer thicknesses");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+
+      // Retrieve the indices of the fields that represent the lithospheric layers.
+      AssertThrow(this->introspection().compositional_name_exists("upper"),ExcMessage("We need a compositional field called 'upper' representing the upper crust."));
+      AssertThrow(this->introspection().compositional_name_exists("lower"),ExcMessage("We need a compositional field called 'lower' representing the lower crust."));
+      AssertThrow(this->introspection().compositional_name_exists("mantle_L"),ExcMessage("We need a compositional field called 'mantle_L' representing the lithospheric part of the mantle."));
+
+      // For now, we assume a 3-layer system with an upper crust, lower crust and lithospheric mantle
```

Comment:
Would it be possible to generalize the plugin to have an infinite number of layers within the lithosphere? I'll go back and look at the Chapman equations to see if this is possible.

Reply:
The equations can also be reformulated such that the surface heat flow and temperature are provided, but the basal temperature is not specified. However, further generalization may be beyond the scope here.